src/services/PromotionService.ts

```diff
@@ -18,117 +19,162 @@ import { isError } from '../utils/api';
 import Routes from '../utils/routes';
 import GooglePlacesService from './GooglePlacesService';
 
-/**
- * Fetches entire list of promotions. If a query object is given, filters the promotions according to the given query.
- * If an error occurs, an empty list will be returned.
- *
- * @param userId [optional] - The userId of a logged in user
- * @param query [optional] - An array of objects with key-value pairs for the query parameters
- */
-export async function getPromotions(
-  userId?: string,
-  query?: GetPromotionDTO
-): Promise<Promotion[]> {
-  const endpoint = Routes.PROMOTIONS.GET;
-  return axios
-    .get(endpoint, {
-      params: {
-        ...query,
-        userId: userId,
-      },
-    })
-    .then(({ data }: AxiosResponse<GetPromotionsResponse>) => {
-      if (isError<GetPromotionsResponse>(data)) {
-        return Promise.reject(data);
-      }
-      return Promise.resolve(data);
-    })
-    .catch((err: Error) => Promise.reject(err));
-}
+class PromotionService {
```

Comment:
Just to be extra sure, you did not change any of the implementations right? Just moved them into a class right

Reply:
Yeah I didn't -- just moved it into a class + added the upvote/downvote ones!